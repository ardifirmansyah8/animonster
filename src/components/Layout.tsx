import { FC, ReactNode } from "react";
import { Link } from "react-router-dom";
import { IconBrandGithubFilled, IconBookmarks } from "@tabler/icons-react";

type Props = {
  children: ReactNode;
};

const Layout: FC<Props> = ({ children }) => {
  return (
    <>
      <header className="flex justify-between z-10 h-14 border-b border-gray-200 p-2 fixed top-0 left-0 w-full bg-white">
        <Link to="/" className="font-bold text-2xl">
          ANIMONSTER
        </Link>
        <div className="flex justify-end gap-2">
          <Link
            to="/bookmarks"
            className="flex items-center p-2 bg-gray-200 border rounded-md"
          >
            <IconBookmarks />
            Bookmarks
          </Link>
          <Link
            to="https://github.com/ardifirmansyah8/animonster"
            target="_blank"
            className="p-2 bg-gray-200 border rounded-md"
          >
            <IconBrandGithubFilled />
          </Link>
        </div>
      </header>
      {children}
    </>
  );
};

export default Layout;
