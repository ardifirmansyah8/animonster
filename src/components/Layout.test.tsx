import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { describe, test, expect } from "vitest";

import Layout from "./Layout";

describe("test Layout", () => {
  test("should render Layout", () => {
    const component = render(
      <MemoryRouter>
        <Layout>test</Layout>
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
