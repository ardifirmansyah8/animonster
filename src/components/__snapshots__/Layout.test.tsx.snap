// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Layout > should render Layout 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="flex justify-between z-10 h-14 border-b border-gray-200 p-2 fixed top-0 left-0 w-full bg-white"
      >
        <a
          class="font-bold text-2xl"
          href="/"
        >
          ANIMONSTER
        </a>
        <div
          class="flex justify-end gap-2"
        >
          <a
            class="flex items-center p-2 bg-gray-200 border rounded-md"
            href="/bookmarks"
          >
            <svg
              class="tabler-icon tabler-icon-bookmarks"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 7a2 2 0 0 1 2 2v12l-5 -3l-5 3v-12a2 2 0 0 1 2 -2h6z"
              />
              <path
                d="M9.265 4a2 2 0 0 1 1.735 -1h6a2 2 0 0 1 2 2v12l-1 -.6"
              />
            </svg>
            Bookmarks
          </a>
          <a
            class="p-2 bg-gray-200 border rounded-md"
            href="https://github.com/ardifirmansyah8/animonster"
            target="_blank"
          >
            <svg
              class="tabler-icon tabler-icon-brand-github-filled"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.315 2.1c.791 -.113 1.9 .145 3.333 .966l.272 .161l.16 .1l.397 -.083a13.3 13.3 0 0 1 4.59 -.08l.456 .08l.396 .083l.161 -.1c1.385 -.84 2.487 -1.17 3.322 -1.148l.164 .008l.147 .017l.076 .014l.05 .011l.144 .047a1 1 0 0 1 .53 .514a5.2 5.2 0 0 1 .397 2.91l-.047 .267l-.046 .196l.123 .163c.574 .795 .93 1.728 1.03 2.707l.023 .295l.007 .272c0 3.855 -1.659 5.883 -4.644 6.68l-.245 .061l-.132 .029l.014 .161l.008 .157l.004 .365l-.002 .213l-.003 3.834a1 1 0 0 1 -.883 .993l-.117 .007h-6a1 1 0 0 1 -.993 -.883l-.007 -.117v-.734c-1.818 .26 -3.03 -.424 -4.11 -1.878l-.535 -.766c-.28 -.396 -.455 -.579 -.589 -.644l-.048 -.019a1 1 0 0 1 .564 -1.918c.642 .188 1.074 .568 1.57 1.239l.538 .769c.76 1.079 1.36 1.459 2.609 1.191l.001 -.678l-.018 -.168a5.03 5.03 0 0 1 -.021 -.824l.017 -.185l.019 -.12l-.108 -.024c-2.976 -.71 -4.703 -2.573 -4.875 -6.139l-.01 -.31l-.004 -.292a5.6 5.6 0 0 1 .908 -3.051l.152 -.222l.122 -.163l-.045 -.196a5.2 5.2 0 0 1 .145 -2.642l.1 -.282l.106 -.253a1 1 0 0 1 .529 -.514l.144 -.047l.154 -.03z"
                fill="currentColor"
                stroke-width="0"
              />
            </svg>
          </a>
        </div>
      </header>
      test
    </div>
  </body>,
  "container": <div>
    <header
      class="flex justify-between z-10 h-14 border-b border-gray-200 p-2 fixed top-0 left-0 w-full bg-white"
    >
      <a
        class="font-bold text-2xl"
        href="/"
      >
        ANIMONSTER
      </a>
      <div
        class="flex justify-end gap-2"
      >
        <a
          class="flex items-center p-2 bg-gray-200 border rounded-md"
          href="/bookmarks"
        >
          <svg
            class="tabler-icon tabler-icon-bookmarks"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 7a2 2 0 0 1 2 2v12l-5 -3l-5 3v-12a2 2 0 0 1 2 -2h6z"
            />
            <path
              d="M9.265 4a2 2 0 0 1 1.735 -1h6a2 2 0 0 1 2 2v12l-1 -.6"
            />
          </svg>
          Bookmarks
        </a>
        <a
          class="p-2 bg-gray-200 border rounded-md"
          href="https://github.com/ardifirmansyah8/animonster"
          target="_blank"
        >
          <svg
            class="tabler-icon tabler-icon-brand-github-filled"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.315 2.1c.791 -.113 1.9 .145 3.333 .966l.272 .161l.16 .1l.397 -.083a13.3 13.3 0 0 1 4.59 -.08l.456 .08l.396 .083l.161 -.1c1.385 -.84 2.487 -1.17 3.322 -1.148l.164 .008l.147 .017l.076 .014l.05 .011l.144 .047a1 1 0 0 1 .53 .514a5.2 5.2 0 0 1 .397 2.91l-.047 .267l-.046 .196l.123 .163c.574 .795 .93 1.728 1.03 2.707l.023 .295l.007 .272c0 3.855 -1.659 5.883 -4.644 6.68l-.245 .061l-.132 .029l.014 .161l.008 .157l.004 .365l-.002 .213l-.003 3.834a1 1 0 0 1 -.883 .993l-.117 .007h-6a1 1 0 0 1 -.993 -.883l-.007 -.117v-.734c-1.818 .26 -3.03 -.424 -4.11 -1.878l-.535 -.766c-.28 -.396 -.455 -.579 -.589 -.644l-.048 -.019a1 1 0 0 1 .564 -1.918c.642 .188 1.074 .568 1.57 1.239l.538 .769c.76 1.079 1.36 1.459 2.609 1.191l.001 -.678l-.018 -.168a5.03 5.03 0 0 1 -.021 -.824l.017 -.185l.019 -.12l-.108 -.024c-2.976 -.71 -4.703 -2.573 -4.875 -6.139l-.01 -.31l-.004 -.292a5.6 5.6 0 0 1 .908 -3.051l.152 -.222l.122 -.163l-.045 -.196a5.2 5.2 0 0 1 .145 -2.642l.1 -.282l.106 -.253a1 1 0 0 1 .529 -.514l.144 -.047l.154 -.03z"
              fill="currentColor"
              stroke-width="0"
            />
          </svg>
        </a>
      </div>
    </header>
    test
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
